### PR TITLE
stage_commit_push_changes to allow empty list items_to_stage

### PR DESCRIPTION
**Pull request by jdw6359**
update implementation of `stage_commit_push_changes` to allow empty list for `items_to_stage` argument

`items_to_stage=[]` argument indicates that there are intentionally no items to stage, and we should not raise a `PyGitOpsError`

### CODE REVIEW

**Comment by jdw6359** [comment]
no changelog present atm, I am trying to decide if we use version `0.12.2` or introduce a new minor version `0.13.0`

File: pygitops/operations.py [hunk 1]
```diff
@@ -35,23 +35,25 @@ def stage_commit_push_changes(
...
         untracked_file_paths = [Path(f) for f in repo.untracked_files]
         items_to_stage = untracked_file_paths + [
             Path(f.a_path) for f in index.diff(None)
         ]
 
-    if not items_to_stage:
-        raise PyGitOpsStagedItemsError(
-            "There are no items to stage, cannot perform commit operation"
-        )
+        if not items_to_stage:
```
Comment (RayOconnor) [inline pygitops/operations.py:53]: **Edit:** After doing some googling it seems like there is not a pythonic AND explicit way to check to see if an array is empty. It seems like `if not my_array:` is correct.

--No Longer Relevant----------------------------------------------------

**Outdated Comment**: when possible it is nice to be explicit about typing expectations. 
When I read `if not items_to_stage:` I infer that the code author expects that `items_to_stage` might be any of `boolean`, `int`, `str`, `array` and wants to handle these cases gracefully by using truthy handle mutiple of these possibilities. 

it seems like we know that `items_to_stage` is an array here for sure. For this reason, I believe it is more clear to use...
`if items_to_stage is empty:`

**Review by RayOconnor — APPROVED** [review]
Thank you for adding this nuanced functionality.

File: pygitops/operations.py (hunk 1 above)
Comment (jdw6359) [reply pygitops/operations.py:53]: `items_to_stage` is type hinted as an `Optional[List[Path]]` so id expect it to be one of the following:

* `None`
* `[]`
* `[path1, ...]`

after doing the `None` check, the `not` check is the most appropriate way to check for an empty array

so, id agree with your edit. thanks for the attention to detail on this!

**Review by CassieNi — APPROVED** [review]

**Review by jhpierce — APPROVED** [review]

**Pull request merged**